tools/lib/optimizeB3dm.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+var Cesium = require('cesium');
+var GltfPipeline = require('gltf-pipeline');
+var Promise = require('bluebird');
+var zlib = require('zlib');
+var isGzipped = require('./isGzipped');
+
+var zlibGunzip = Promise.promisify(zlib.gunzip);
+
+var Cartesian3 = Cesium.Cartesian3;
+var DeveloperError = Cesium.DeveloperError;
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+
+var Pipeline = GltfPipeline.Pipeline;
+var addCesiumRTC = GltfPipeline.addCesiumRTC;
+var getBinaryGltf = GltfPipeline.getBinaryGltf;
+var parseBinaryGltf = GltfPipeline.parseBinaryGltf;
+
+module.exports = optimizeB3dm;
+
+/**
+ * Given an input, possibly gzipped buffer containing a b3dm, optimize it using gltf-pipeline with the provided options
```

Comment:
I think it would make more sense for the potential unzip to happen the bin file instead.